### ISSUE TITLE
Doesn't build on Sierra (GM) with Xcode 8

### ISSUE THREAD

**Issue by chdiza** [jhford/screenresolution#25]
On the Sierra GM with Xcode 8 and the CLT for Xcode 8, building from the latest commit results in the following:

```
sed -e "s/@VERSION@/\"1.7dev\"/" < version-tmpl.h > version.h
clang   -arch i386 -arch x86_64 main.c -c -o main.o
clang   -arch i386 -arch x86_64 cg_utils.c -c -o cg_utils.o
clang   -arch i386 -arch x86_64 -framework Foundation -framework ApplicationServices main.o cg_utils.o -o screenresolution
Undefined symbols for architecture x86_64:
  "_MIN", referenced from:
      _listAvailableModes in main.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [screenresolution] Error 1
```

Prior to the failure point, there are also warnings emitted:

```
sed -e "s/@VERSION@/\"1.7dev\"/" < version-tmpl.h > version.h
clang   -arch i386 -arch x86_64 main.c -c -o main.o
main.c:175:26: warning: implicit declaration of function 'MIN' is invalid in C99
      [-Wimplicit-function-declaration]
        idxDisplayMode = MIN(idxDisplayMode, numModes - 1);
                         ^
1 warning generated.
main.c:175:26: warning: implicit declaration of function 'MIN' is invalid in C99
      [-Wimplicit-function-declaration]
        idxDisplayMode = MIN(idxDisplayMode, numModes - 1);
                         ^
1 warning generated.
clang   -arch i386 -arch x86_64 cg_utils.c -c -o cg_utils.o
cg_utils.c:25:30: warning: 'CGDisplayModeCopyPixelEncoding' is deprecated: first
      deprecated in macOS 10.11 [-Wdeprecated-declarations]
        CFStringRef pixelEncoding = CGDisplayModeCopyPixelEncoding(mode);
                                    ^
/System/Library/Frameworks/CoreGraphics.framework/Headers/CGDirectDisplay.h:174:34: note:
      'CGDisplayModeCopyPixelEncoding' has been explicitly marked deprecated here
CG_EXTERN CFStringRef __nullable CGDisplayModeCopyPixelEncoding(
                                 ^
1 warning generated.
cg_utils.c:25:30: warning: 'CGDisplayModeCopyPixelEncoding' is deprecated: first
      deprecated in macOS 10.11 [-Wdeprecated-declarations]
        CFStringRef pixelEncoding = CGDisplayModeCopyPixelEncoding(mode);
                                    ^
/System/Library/Frameworks/CoreGraphics.framework/Headers/CGDirectDisplay.h:174:34: note:
      'CGDisplayModeCopyPixelEncoding' has been explicitly marked deprecated here
CG_EXTERN CFStringRef __nullable CGDisplayModeCopyPixelEncoding(
                                 ^
1 warning generated.
```

When building from the 1.6 release tarball, I get the warnings but not the build failure.

**Comment by chdiza** [comment]
The same error happens also on El Capitan with Xcode 8, so it's not a Sierra-only problem.

**Comment by mistydemeo** [comment]
Fixed in #26.

**Comment by hantuzun** [comment]
I get a similar build from the current master `9ee58cfb088e72149f6e244517fcb3d6dfd813c5`:

```sh
$ make dmg
clang   -arch i386 -arch x86_64 -framework Foundation -framework ApplicationServices main.o cg_utils.o -o screenresolution
Undefined symbols for architecture i386:
  "_MIN", referenced from:
      _listAvailableModes in main.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [screenresolution] Error 1
```

I'm running High Sierra (10.13.1 -- 17B1003).

**Comment by avmaint** [comment]
same problem as hantuzun on 10.12.16 Sierra.

**Comment by shaunc** [comment]
Just to confirm on high Sierra fix in #26 works

**Comment by jjarava** [comment]
In High Sierra, XCode Version 10.1 (10B61), applying fix #26 makes it build and I get a binary and a .dmg so this fixes the issue

Now to see if it _works_

**Comment by mistydemeo** [comment]
Thanks so much for fixing this! Much appreciated. Would you be able to tag a new release?
